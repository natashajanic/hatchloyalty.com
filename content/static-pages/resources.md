---
title: "Guides and Tutorials"
---

# Tutorials

End-to-end real-world examples of using Hatch to support the marketing program as a whole, as well as support specific initiatives.

## [Create a Behavioral Offer](/resource/behavioral-offer)

How to compile and configure a complex behavioral marketing initiative with advanced custom segmentation in Hatch.

## [Create a Purchase-based Offer](/resource/purchase-based-offer)

How to compile and configure a complex purchase-based marketing initiative in Hatch.

## [Enroll Members](/resource/member-enrollment)

How to acquire new Memberships in your program with a tailored onboarding.

## [Process Transactions](/resource/transaction-processing)

Identify a Membership and record their activity at your store, online, or at other point-of-sale devices.

## [Redeem Rewards](/resource/reward-redemption)

Present available Rewards and allow them to redeem them with the points they've accumulated.


# Guides

Guides on how Hatch uses specific concepts to execute your loyalty program.

## Program Configuration

### [Product Groups](/resource/product-groups)

Classify products for use in setting up Transaction-based Offers.

### [Custom Segments](/resource/custom-segments)

Enable custom segmentation based on your own business intelligence systems.

## Consumer Behavior

### [Offers](/resource/offers)

The primary tool to encourage a desired change in consumer behavior. Offers are composed of five components:

- **[Earning Mechanism](/resource/offers/earning-mechanisms)**: The behavior that the marketing initiative promotes, as well as the incentive given to a member for engaging with it, e.g., the purchase of a specific product in exchange for points.
- **[Constraints](/resource/offers/constraints)**: Any pre-fulfillment actions or conditions that must be met in order for a member to engage with the Offer, e.g., "activating" the Offer.
- **[Audiences](/resource/offers/audiences)**: Any factors that dictate what subset of Memberships should be shown the Offer, e.g., that the Membership is in a required [custom segment](/resource/custom-segments).
- **[Limits](/resource/offers/limits)**: Any limits, either global or per-member, that fulfillment or engagement with the Offer must abide by, e.g., a limit of 4000 fulfillments total.
- **[Marketing Contents](/resource/offers/marketing-contents)**: The locale-specific marketing copy that is shown to a member in a mobile application or digital signage, e.g., the title and an image.

### [Rewards](/resource/rewards)

The tangible benefits that a member can purchse with their accumulated points. Rewards are composed of two components:

- **[Marketing Contents](/resource/rewards/marketing-contents)**: The locale-specific marketing copy that is shown to a member in a mobile application or digital signage, e.g., the title and an image.
- **[Segmentation](/resource/rewards/segmentation)**: The ability to target a subset of Memberships with a personalized Reward.