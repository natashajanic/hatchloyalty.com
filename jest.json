{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
