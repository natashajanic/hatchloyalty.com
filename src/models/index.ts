import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
