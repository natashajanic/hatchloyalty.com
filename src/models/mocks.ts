import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
