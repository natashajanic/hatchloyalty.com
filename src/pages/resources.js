import React from 'react'
import { graphql, Link } from 'gatsby'
import { BookOpen } from 'react-feather'
import Layout from '../components/layout'
import SEO from '../components/seo'
import Text from '../components/text'
import Box from '../components/box'
import BlogPostContainer from '../components/blog-post-container'
import IconCircle from '../components/icon-circle'

class DocsIndex extends React.Component {
  render() {
    const { data } = this.props;
    const resources = data.allMarkdownRemark.edges

    return (
      <Layout pageStyle="offWhite">
        <SEO title="All Resources" keywords={['hatch', 'hatch loyalty', 'loyalty', 'docs', 'personalization', 'activation',]} />
        <BlogPostContainer
          is="main"
        >
          <Box display="flex" alignItems="center">
            <IconCircle bg="green">
              <BookOpen size={20} color="#fff" />
            </IconCircle>
            <Text fontSize={1} ml={2}>Hatch Resources/Docs</Text>
          </Box>

          <Text is="h2" fontSize={5}>Learn from experts. Hatch is here to help you build stronger relationships with your customers.</Text>

          {resources.map(({ node }) => {
            const title = node.frontmatter.title || node.fields.slug
            return (
              <div
                key={node.fields.slug}
              >

                <Link
                  to={node.fields.slug}
                  style={{
                    color: '#666666',
                    display: 'block',
                    textDecoration: 'none'
                  }}
                >
                  {title}
                </Link>

                <Text
                  is="p"
                  color="grayDark"
                  fontSize={3}
                  dangerouslySetInnerHTML={{ __html: node.excerpt }}
                />

                <Link
                  to={node.fields.slug}
                  color="grayDark"
                >
                  Read More
                </Link>
              </div>
            )
          })}
        </BlogPostContainer>
      </Layout>
    )
  }
}

export default DocsIndex

export const docIndexQuery = graphql`
  query {
    allMarkdownRemark(
      sort: { fields: [frontmatter___date], order: DESC }
      filter: { fields: { sourceName: { eq: "resources"}}},

    ) {
      edges {
        node {
          excerpt
          fields {
            slug
          }
          frontmatter {
            title
          }
        }
      }
    }
  }
`
