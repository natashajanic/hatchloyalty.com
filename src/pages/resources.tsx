import * as React from 'react'
import { graphql, Link } from 'gatsby'
import { BookOpen } from 'react-feather'
import Layout from 'src/components/layout'
import SEO from 'src/components/seo'
import Text from 'src/components/text'
import Box from 'src/components/box'
import BlogPostContainer from 'src/components/blog-post-container'
import IconCircle from 'src/components/icon-circle'

interface IResourcesPageProps {
  data: {
    allMarkdownRemark: {
      edges: Array<{
        node: {
          excerpt: string
          fields: { slug: string }
          frontmatter: {
            title: string
          }
        }
      }>
    }
  }
}

class ResourcesPage extends React.Component<IResourcesPageProps, {}> {
  render() {
    const { data } = this.props
    const resources = data.allMarkdownRemark.edges.map(({ node }) => {
      const title = node.frontmatter.title || node.fields.slug
      return (
        <div key={node.fields.slug}>
          <Link
            to={`/resource${node.fields.slug}`}
            style={{
              color: '#666666',
              display: 'block',
              textDecoration: 'none',
            }}
          >
            {title}
          </Link>

          <Text
            is="p"
            color="grayDark"
            fontSize={3}
            dangerouslySetInnerHTML={{ __html: node.excerpt }}
          />

          <Link to={`/resource${node.fields.slug}`} color="grayDark">
            Read More
          </Link>
        </div>
      )
    })

    return (
      <Layout pageStyle="offWhite">
        <SEO
          title="All Resources"
          keywords={[
            'hatch',
            'hatch loyalty',
            'loyalty',
            'docs',
            'personalization',
            'activation',
          ]}
        />
        <BlogPostContainer is="main">
          <Box display="flex" alignItems="center">
            <IconCircle bg="green">
              <BookOpen size={20} color="#fff" />
            </IconCircle>
            <Text fontSize={1} ml={2}>
              Hatch Resources/Docs
            </Text>
          </Box>

          <Text is="h2" fontSize={5}>
            Learn from experts. Hatch is here to help you build stronger
            relationships with your customers.
          </Text>

          {resources}
        </BlogPostContainer>
      </Layout>
    )
  }
}

export default ResourcesPage

export const pageQuery = graphql`
  query {
    allMarkdownRemark(
      sort: { fields: [frontmatter___date], order: DESC }
      filter: { fields: { sourceName: { eq: "resources" } } }
    ) {
      edges {
        node {
          excerpt
          fields {
            slug
          }
          frontmatter {
            title
          }
        }
      }
    }
  }
`
