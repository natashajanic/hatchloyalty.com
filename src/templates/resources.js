import React from 'react'
import { graphql } from 'gatsby'
import system from 'system-components'
import Layout from '../components/layout'
import Box from '../components/box'
import Wrapper from '../components/wrapper'
import SEO from '../components/seo'

const PostTitle = system({
  is: 'h1',
  borderBottom: '1px solid',
  borderColor: 'grayLight',
  fontSize: 5,
  m: 0,
  pb: 4,
})

class ResourceTemplate extends React.Component {
  render() {
    const post = this.props.data.markdownRemark
    return (
      <Layout>
        <SEO title={post.frontmatter.title} description={post.excerpt} />
        <Wrapper>
          <Box py={4}>
            <PostTitle>{post.frontmatter.title}</PostTitle>
          </Box>

          <div dangerouslySetInnerHTML={{ __html: post.html }} />
        </Wrapper>
      </Layout>
    )
  }
}

export default ResourceTemplate

export const pageQuery = graphql`
  query ResourceBySlug($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      id
      excerpt(pruneLength: 160)
      html
      frontmatter {
        title
      }
    }
  }
`