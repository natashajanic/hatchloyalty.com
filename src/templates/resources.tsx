import * as React from 'react'
import { graphql } from 'gatsby'
import system from 'system-components'
import Layout from '../components/layout'
import Box from '../components/box'
import Wrapper from '../components/wrapper'
import SEO from '../components/seo'

const PostTitle = system({
  is: 'h1',
  borderBottom: '1px solid',
  borderColor: 'grayLight',
  fontSize: 5,
  m: 0,
  pb: 4,
})

interface IResourceTemplateProps {
  data: {
    markdownRemark: {
      excerpt: string
      frontmatter: {
        title: string
      }
      html: string
    }
  }
}

class ResourceTemplate extends React.Component<IResourceTemplateProps, {}> {
  render() {
    const post = this.props.data.markdownRemark
    return (
      <Layout>
        <SEO title={post.frontmatter.title} description={post.excerpt} />
        <Wrapper>
          <Box py={4}>
            <PostTitle>{post.frontmatter.title}</PostTitle>
          </Box>

          <div dangerouslySetInnerHTML={{ __html: post.html }} />
        </Wrapper>
      </Layout>
    )
  }
}

export default ResourceTemplate

export const pageQuery = graphql`
  query ResourceBySlug($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      excerpt(pruneLength: 160)
      html
      frontmatter {
        title
      }
    }
  }
`
