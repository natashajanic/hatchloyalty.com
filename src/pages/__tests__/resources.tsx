import { shallow, ShallowWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Text from 'src/components/text'
import ResourcePostList from 'src/components/resource-post-list'
import { mockResourcePost } from 'src/models/mocks'
import { PureResourcesPage, IPureResourcesPageProps } from '../resources'

describe('ResourcesPage', () => {
  let props: IPureResourcesPageProps
  let mountedComponent: ShallowWrapper<any, any, any>

  beforeEach(() => {
    props = {
      data: {
        allMarkdownRemark: {
          edges: [
            {
              node: {
                ...mockResourcePost,
                frontmatter: { title: 'Test Post 1' },
              },
            },
            {
              node: {
                ...mockResourcePost,
                frontmatter: { title: 'Test Post 2' },
              },
            },
            {
              node: {
                ...mockResourcePost,
                frontmatter: { title: 'Test Post 3' },
              },
            },
          ],
        },
      },
    }
    mountedComponent = shallow(<PureResourcesPage {...props} />)
  })

  describe('render', () => {
    it('displays a page title', async () => {
      expect(mountedComponent.find(Text).length).toEqual(2)
      const titleText = mountedComponent.find(Text).first()
      expect(titleText.text()).toEqual('Hatch Resources/Docs')
    })

    it('displays the list of resource posts', async () => {
      expect(mountedComponent.find(ResourcePostList).length).toEqual(1)
      expect(
        mountedComponent.find(ResourcePostList).prop('posts')
      ).toMatchObject([
        props.data.allMarkdownRemark.edges[0].node,
        props.data.allMarkdownRemark.edges[1].node,
        props.data.allMarkdownRemark.edges[2].node,
      ])
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
