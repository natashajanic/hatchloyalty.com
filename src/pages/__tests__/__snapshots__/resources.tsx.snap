// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesPage render matches the expected snapshot 1`] = `
<Layout
  pageStyle="offWhite"
>
  <SEO
    keywords={
      Array [
        "hatch",
        "hatch loyalty",
        "loyalty",
        "docs",
        "personalization",
        "activation",
      ]
    }
    title="All Resources"
  />
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "maxWidth",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    is="main"
    m="0 auto"
    maxWidth="1024px"
    px={
      Array [
        3,
        4,
      ]
    }
    py={5}
  >
    <Styled(Component)
      alignItems="center"
      blacklist={
        Array [
          "css",
          "alignItems",
          "border",
          "borderColor",
          "borderBottom",
          "borderRadius",
          "color",
          "bg",
          "display",
          "justifyContent",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "width",
        ]
      }
      display="flex"
    >
      <Styled(Component)
        bg="green"
        blacklist={
          Array [
            "css",
            "borderRadius",
            "display",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "color",
            "bg",
            "justifyContent",
            "width",
          ]
        }
        borderRadius={100}
        display="inline-flex"
        p={2}
      >
        <BookOpen
          color="#fff"
          size={20}
        />
      </Styled(Component)>
      <Text
        blacklist={
          Array [
            "css",
            "fontSize",
            "lineHeight",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "textStyle",
            "alignItems",
            "color",
            "bg",
            "display",
            "flex",
            "fontWeight",
            "textAlign",
          ]
        }
        fontSize={1}
        lineHeight="1.5"
        m={0}
        ml={2}
        textStyle="normal"
      >
        Hatch Resources/Docs
      </Text>
    </Styled(Component)>
    <Text
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      fontSize={5}
      is="h2"
      lineHeight="1.5"
      m={0}
      textStyle="normal"
    >
      Learn from experts. Hatch is here to help you build stronger relationships with your customers.
    </Text>
    <ResourcePostList
      posts={
        Array [
          Object {
            "excerpt": "This is the excerpt for a resource post.",
            "fields": Object {
              "slug": "/test-resource-post/",
            },
            "frontmatter": Object {
              "title": "Test Post 1",
            },
            "html": "<div>Test resource post body content.</div>",
          },
          Object {
            "excerpt": "This is the excerpt for a resource post.",
            "fields": Object {
              "slug": "/test-resource-post/",
            },
            "frontmatter": Object {
              "title": "Test Post 2",
            },
            "html": "<div>Test resource post body content.</div>",
          },
          Object {
            "excerpt": "This is the excerpt for a resource post.",
            "fields": Object {
              "slug": "/test-resource-post/",
            },
            "frontmatter": Object {
              "title": "Test Post 3",
            },
            "html": "<div>Test resource post body content.</div>",
          },
        ]
      }
    />
  </Styled(Component)>
</Layout>
`;
